shell/common/rasterizer.cc

```diff
@@ -265,9 +268,36 @@ RasterStatus Rasterizer::DoDraw(
   // TODO(liyuqian): in Fuchsia, the rasterization doesn't finish when
   // Rasterizer::DoDraw finishes. Future work is needed to adapt the timestamp
   // for Fuchsia to capture SceneUpdateContext::ExecutePaintTasks.
-  timing.Set(FrameTiming::kRasterFinish, fml::TimePoint::Now());
+  const auto raster_finish_time = fml::TimePoint::Now();
+  timing.Set(FrameTiming::kRasterFinish, raster_finish_time);
   delegate_.OnFrameRasterized(timing);
 
+  if (raster_finish_time > frame_target_time) {
+    fml::TimePoint latest_frame_target_time =
+        delegate_.GetLatestFrameTargetTime();
+    const auto frame_budget_millis = delegate_.GetFrameBudget().count();
+    if (latest_frame_target_time < raster_finish_time) {
+      latest_frame_target_time =
+          latest_frame_target_time +
+          fml::TimeDelta::FromMilliseconds(frame_budget_millis);
```

Comment:
Does this work? `frame_budget_millis` is, I think, a double representing the number of millis. The `FromMilliseconds()` method takes an int64_t, doesn't it? I'm surprised this doesn't generate an error, but I can't find a `FromMilliseconds(double)` method. There is a `FromSecondsF(double)` method that could be used to maintain precision. I'd add a `FromMillisecondsF(double)` factory or use the "seconds" version here to maintain accuracy.